frontend/src/components/RequireAuth.tsx

```diff
@@ -1,9 +1,13 @@
 import {ReactElement, useContext} from "react";
 import {AuthContext} from "../context/AuthProvider";
 import {Navigate} from "react-router-dom";
+import {ILoginController} from "../models/ControllerTypes";
+import LoginController from "../controllers/LoginController";
 
 export default function RequireAuth({children}: { children: ReactElement<any, any> }) {
-    const {token} = useContext(AuthContext)
+    const {token, logout} = useContext(AuthContext)
+    const controller: ILoginController = LoginController()
+    controller.checkLoggedIn().catch( () => logout());
```

Comment:
I would not ask the backend  if your still logged in. 
I does not make anything more secure,
You are making a lot of request to your backend,
every time changing the route the user must wait for the request to finish,

You could check the expireationTime of the token locally